src/main/java/org/opentripplanner/model/plan/legreference/LegReferenceSerializer.java

```diff
@@ -19,6 +25,17 @@
 public class LegReferenceSerializer {
 
   private static final Logger LOG = LoggerFactory.getLogger(LegReferenceSerializer.class);
+  private static final DateTimeFormatter LENIENT_ISO_LOCAL_DATE = new DateTimeFormatterBuilder()
+    .appendValue(YEAR, 4)
+    .optionalStart()
+    .appendLiteral('-')
+    .optionalEnd()
+    .appendValue(MONTH_OF_YEAR, 2)
+    .optionalStart()
+    .appendLiteral('-')
+    .optionalEnd()
+    .appendValue(DAY_OF_MONTH, 2)
+    .toFormatter();
```

Comment:
So this is to support optional '-' in the input? If so should there be a comment that this can be replaced with  `DateTimeFormatter.ISO_LOCAL_DATE` after the next release of OTP?

Reply:
Done